app/assets/javascripts/admin/addon/controllers/admin-site-settings.js

```diff
@@ -2,65 +2,32 @@ import Controller from "@ember/controller";
 import { action } from "@ember/object";
 import { alias } from "@ember/object/computed";
 import { service } from "@ember/service";
-import { isEmpty } from "@ember/utils";
 import { debounce } from "discourse/lib/decorators";
 import { INPUT_DELAY } from "discourse/lib/environment";
-import SiteSettingFilter from "admin/lib/site-setting-filter";
 
 export default class AdminSiteSettingsController extends Controller {
   @service router;
 
-  @alias("model") allSiteSettings;
-
-  filter = "";
-  visibleSiteSettings = null;
-  siteSettingFilter = null;
-
-  filterContentNow(filterData, category) {
-    this.siteSettingFilter ??= new SiteSettingFilter(this.allSiteSettings);
-
-    if (isEmpty(this.allSiteSettings)) {
-      return;
-    }
-
-    this.set("filter", filterData.filter);
-
-    if (isEmpty(filterData.filter) && !filterData.onlyOverridden) {
-      this.set("visibleSiteSettings", this.allSiteSettings);
-      if (this.categoryNameKey === "all_results") {
-        this.router.transitionTo("adminSiteSettings");
-      }
-      return;
-    }
-
-    const matchesGroupedByCategory = this.siteSettingFilter.filterSettings(
-      filterData.filter,
-      { onlyOverridden: filterData.onlyOverridden }
-    );
-
-    const categoryMatches = matchesGroupedByCategory.findBy(
-      "nameKey",
-      category
-    );
-
-    if (!categoryMatches || categoryMatches.count === 0) {
-      category = "all_results";
-    }
-
-    this.set("visibleSiteSettings", matchesGroupedByCategory);
-    this.router.transitionTo(
-      "adminSiteSettingsCategory",
-      category || "all_results"
-    );
-  }
+  @alias("model.filteredSettings") visibleSiteSettings;
+  @alias("model.filtersApplied") filtersApplied;
 
   @debounce(INPUT_DELAY)
   filterContent(filterData) {
     if (this._skipBounce) {
       this.set("_skipBounce", false);
     } else {
-      if (!this.isDestroyed) {
-        this.filterContentNow(filterData, this.categoryNameKey);
+      const currentParams = this.router.currentRoute.queryParams;
+      const queryParams = {
+        filter: filterData.filter || undefined,
+        onlyOverridden: filterData.onlyOverridden || undefined,
+      };
+
+      const paramsChanged =
+        currentParams.filter !== queryParams.filter ||
+        currentParams.onlyOverridden !== queryParams.onlyOverridden;
+
+      if (!this.isDestroyed && paramsChanged) {
```

Comment:
do we need the `this.isDestroyed` here? I thought the `@debounce` macro would deal with this as it's supposed to be cancel on destroy

Reply:
It was here originally. So it's either keep or remove blindly risking regression. Since I'm off two weeks I chose the former. 😬